templates/rustdoc/head.html

```diff
@@ -3,4 +3,8 @@
 
         <link rel="search" href="/-/static/opensearch.xml" type="application/opensearchdescription+xml" title="Docs.rs" />
 
+        {%- if canonical_url -%}
+        <link rel="canonical" href="{{canonical_url | safe}}" />
```

Comment:
@jyn514 wouldn't we need the same logic for other versioned pages? The docs.rs crate page comes to mind

Reply:
IMO we shouldn't bother. Almost no one searches for the crate page, so it's not that critical to add canonicalization for it. If we find down the line that people are looking for that page more often, we can always add canonicalization later.